tracker/utils.py

```diff
@@ -317,3 +322,49 @@ def attach_link_to_issue(issue_title: str, issue_link: str) -> str:
     title = f'<a href="{issue_link}">{issue_title}</a>'
     return title
 
+
+def get_repository_from_issue(issue: dict) -> dict:
+    repository_url = issue.get("repository_url", "")
+    if repository_url:
+        parts = repository_url.rstrip("/").split("/")
+        return {"author": parts[-2], "name": parts[-1]}
+    return {}
+
+
+def get_time_before_deadline(issue: dict) -> str:
+    """
+    Returns the time remaining before the deadline of an assigned issue.
+    If the issue has no assignee or deadline, returns appropriate messages.
+
+    :param issue: The issue dictionary containing information about the issue.
+    :return: Time remaining in a human-readable format.
+    """
+
+    assignment_info = check_issue_assignment_events(issue)
+    assigned_at = assignment_info.get("assigned_at")
+
+    if not assigned_at:
+        return "This issue is not assigned."
+
+    repository_details = get_repository_from_issue(issue)
+    if not repository_details:
+        return "Repository details not found."
+
+    from .models import Repository
+
+    repo = Repository.objects.get(
+        author=repository_details["author"], name=repository_details["name"]
+    )
+    time_limit_seconds = repo.time_limit
+
+    assigned_time = datetime.strptime(assigned_at, "%Y-%m-%dT%H:%M:%SZ").replace(
+        tzinfo=timezone.utc
+    )
+    deadline_datetime = assigned_time + timedelta(seconds=time_limit_seconds)
+    now = datetime.now(timezone.utc)
+
+    if deadline_datetime > now:
+        remaining_time = deadline_datetime - now
+        return f"Time remaining: {remaining_time.days} days, {remaining_time.seconds // 3600} hours"
```

Comment:
Move `3600` magic number to `values.py` and import it from there

Reply:
Updated!